src/dmd/nspace.d

```diff
@@ -1,7 +1,44 @@
 /**
- * Compiler implementation of the
- * $(LINK2 http://www.dlang.org, D programming language).
+ * A scoped C++ namespace symbol
  *
+ * D supports the following syntax to declare symbol(s) as being part of a
+ * C++ namespace:
+ * ---
+ * extern (C++, "myNamespace") { /+ Symbols +/ }       // String variant
+ * extern (C++, SomeNamespace) { /+ Other symbols +/ } // Identifier variant
+ * ---
+ * The first form is an attribute and only affects mangling, and is implemented
+ * in `dmd.attrib`.
+ * The second form introduces a named scope and allows symbols to be refered
+ * to with or without the namespace name, much like a named template mixin,
+ * and is implemented in this module.
+ * ---
+ * extern (C++, Basket)
+ * {
+ *     struct StrawBerry;
+ *     void swapFood (Strawberry* f1, Strawberry* f2);
+ * }
+ * void main ()
+ * {
+ *     Basket.StrawBerry fruit1;
+ *     StrawBerry fruit2;
+ *     Basket.swapFood(fruit1, fruit2);
+ *     swapFood(fruit1, fruit2);
+ * }
+ * ---
+ * Hence the `Nspace` symbol implements the usual `ScopeDsymbol` semantics.
+ *
+ * Note that it implies `extern(C++)` so it cannot be used as a generic
+ * named scope. Additionally, `Nspace` with the same `Identifier` can be
+ * defined in different module (as C++ allows a namespace to be spread accross
+ * translation units), but symbols in it should be considered
+ * part of the same scope. Lastly, not all possible C++ namespace names
+ * are valid D identifier.
+ *
+ * Those shortcomings led to the development of the string variant
+ * (`dmd.attrib : CPPNamespaceDeclaration`), which should be preferred.
```

Comment:
I don't agree with the last two lines. Please stop editorializing this.

Reply:
Yes, I know your opinion on this, and you know mine.

Reply:
Well, those shortcomings *did* lead to the development of the string variant, that's not an opinion. I'll remove the "should be preferred" though.
And I still strongly think it should be deprecated. It's been ~16 months and I haven't seen a single person happy with this feature, and none of the bindings in druntime use it.

Reply:
Actually removed the whole sentence, the previous paragraph should be enough.

Reply:
thank you